src/Components/DataHistoryExplorer/Internal/TimeSeriesViewer/Internal/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -106,6 +113,16 @@ const TimeSeriesChart: React.FC<ITimeSeriesChartProps> = (props) => {
             onChartOptionsChange(chartOptions);
         }
     }, [chartOptions]);
+    useEffect(() => {
```

Comment:
it feels weird that we need to do this same logic in the chart as the table component. Could we instead feed the data into those components and do this in the parent? That way we don't have both components fetching data separately? That seems like a good way to end up with the chart doing some different manipulation than the table and they can get out of sync.

Reply:
Yeah since both Chart and Table is children of TimeSeriesViewer which needs to pass this missing data information to TimeSeriesBuilder on the left to show the triange noData icon for individual series, I needed to bubble up series with no data from places where I do data fetching, in which case it is both Chart and Table. After I make the refactor there will be no need for this, wait for the following PR please...